### ISSUE TITLE
Should document `cue.mod/module.cue` and module vs package

### ISSUE THREAD

**Issue by jpadams** [dagger/dagger#2186]
### What is the issue?

After a `dagger project init` (without args), the created `cue.mod/module.cue` file contains:

```
module: ""
```

It's not clear to a new user what that means and if it's related to the package name in their cue files.

https://cuelang.org/docs/concepts/packages/#packages
https://cuelang.org/docs/concepts/packages/#modules
https://cuelang.org/docs/tutorials/tour/packages/packages/
https://cuelang.org/docs/tutorials/tour/packages/imports/
https://github.com/dagger/dagger/pull/1685

It's also probably not clear to new users when they need to create a module name and how it relates to package name when building a dagger package destined to be shared (often via github, etc).

https://github.com/dagger/dagger/issues/2175

**Comment by morlay** [comment]
it is old issue of cue

once official mod manager support, it goes better.

https://github.com/cue-lang/cue/issues/851


I created a simple one for now.
https://github.com/octohelm/cuemod

**Comment by helderco** [comment]
**TLDR: The use of a module is optional, but required if one wants to import files from it.**

Documentation for this is being worked on:
- https://github.com/dagger/dagger/pull/2271

### Definitions

In summary:
- **CUE Package:** directory with cue files, each including a `package` definition on top, making it importable (e.g, `universe.dagger.io/docker` is a package, `universe.dagger.io/docker/cli` is another package);
- **CUE Module:** directory with a `cue.mod` directory which makes for the prefix/root of importable packages (e.g, `universe.dagger.io` is a module;

### Anonymous module (default)

The default module name is an empty string, which means it's anonymous. That's when you **don't need to import other packages inside your module**. It represents most cases where someone has a dagger plan and imports everything from third-party packages like _universe_.

### When to use a module

You can add a different module name, if you want to **import other packages from inside your module**. This is required so you have a prefix/root before your imports. This is when you want to split your code into multiple files.

**The module path becomes the import path prefix for packages in the module.** This might be the name of a domain you own or another name you control (such as your company name), even your email, followed **optionally** by a descriptive path (e.g., project name).

Example:

```
root                    // must contain:
|-- cue.mod
|   |-- module.cue      // module: "example.com/pkg"
|-- schemas
|   |-- compose
|   |   |-- spec.cue   // package compose
...
|-- plan.cue            // import "example.com/pkg/schemas/compose"
```
